src/test/java/com/amazon/opendistroforelasticsearch/sql/unittest/expression/core/UnaryExpressionTest.java

```diff
@@ -0,0 +1,137 @@
+import static com.amazon.opendistroforelasticsearch.sql.expression.model.ExprValueFactory.stringValue;
+import static org.junit.Assert.assertEquals;
+
+@RunWith(MockitoJUnitRunner.class)
+public class UnaryExpressionTest extends ExpressionTest {
+
+    @Rule
+    public ExpectedException exceptionRule = ExpectedException.none();
+
+    @Test
+    public void absShouldPass() {
+        assertEquals(2.0d, apply(ScalarOperation.ABS, literal(doubleValue(-2d))));
+    }
+
+    @Test
+    public void asinShouldPass() {
+        assertEquals(0.1001674211615598d, apply(ScalarOperation.ASIN, literal(doubleValue(0.1d))));
+    }
+
+    @Test
+    public void atanShouldPass() {
+        assertEquals(1.1071487177940904d, apply(ScalarOperation.ATAN, literal(doubleValue(2d))));
+    }
+
+    @Test
+    public void tanShouldPass() {
+        assertEquals(-2.185039863261519, apply(ScalarOperation.TAN, literal(doubleValue(2d))));
+    }
+
+    @Test
+    public void atan2ShouldPass() {
+        assertEquals(1.1071487177940904d,
+                     apply(ScalarOperation.ATAN2, literal(doubleValue(2d)), literal(doubleValue(1d))));
+    }
+
+    @Test
+    public void cbrtShouldPass() {
+        assertEquals(1.2599210498948732d,
+                     apply(ScalarOperation.CBRT, literal(doubleValue(2d))));
+    }
+
+    @Test
+    public void ceilShouldPass() {
+        assertEquals(3.0d,
+                     apply(ScalarOperation.CEIL, literal(doubleValue(2.1d))));
+    }
+
+    @Test
+    public void floorShouldPass() {
+        assertEquals(2.0d,
+                     apply(ScalarOperation.FLOOR, literal(doubleValue(2.1d))));
+    }
+
+    @Test
+    public void cosShouldPass() {
+        assertEquals(-0.4161468365471424d,
+                     apply(ScalarOperation.COS, literal(doubleValue(2d))));
+    }
+
+    @Test
+    public void coshShouldPass() {
+        assertEquals(3.7621956910836314d,
+                     apply(ScalarOperation.COSH, literal(doubleValue(2d))));
+    }
+
+    @Test
+    public void expShouldPass() {
+        assertEquals(7.38905609893065d,
+                     apply(ScalarOperation.EXP, literal(doubleValue(2d))));
+    }
+
+    @Test
+    public void lnShouldPass() {
+        assertEquals(0.6931471805599453d,
+                     apply(ScalarOperation.LN, literal(doubleValue(2d))));
+    }
+
+    @Test
+    public void logShouldPass() {
+        assertEquals(0.6931471805599453d,
+                     apply(ScalarOperation.LOG, literal(doubleValue(2d))));
+    }
+
+    @Test
+    public void log2ShouldPass() {
+        assertEquals(1.0d,
+                     apply(ScalarOperation.LOG2, literal(doubleValue(2d))));
+    }
+
+    @Test
+    public void log10ShouldPass() {
+        assertEquals(0.3010299956639812,
+                     apply(ScalarOperation.LOG10, literal(doubleValue(2d))));
+    }
+
+    @Test
+    public void absWithStringShouldThrowException() {
```

Comment:
Any specific reason only ABS() and Tan2() were tested against with String values?

Reply:
ideally all the type should been tested. the purpose of the test is to verify the error message is as expected.